src/main/java/org/folio/innreach/external/dto/PatronInfoResponse.java

```diff
@@ -24,7 +24,7 @@ public static PatronInfoResponse of(PatronInfo patronInfo, Boolean requestAllowe
     return PatronInfoResponse.builder()
       .status(OK_STATUS)
       .reason(OK_REASON)
-      .patronInfo(patronInfo)
+      .patronInfo(requestAllowed ? patronInfo : null)
```

Comment:
if patronInfo is not needed when request is not allowed then we don't need to prepare `PatronInfo` object. Lets move this check to service without calling `getPatronInfo(...);` if the statement is true